src/services/accounts/AccountsValidateService.ts

```diff
@@ -42,16 +49,29 @@ export class AccountsValidateService extends AbstractService {
 
 		if (defaults.allowedEncodedLengths.includes(u8Address.length)) {
 			const [isValid, , , ss58Prefix] = checkAddressChecksum(u8Address);
-
+			network = null;
+			for (const networkParams of allNetworks) {
+				if (networkParams['prefix'] === ss58Prefix) {
+					network = networkParams['network'];
+					break;
+				}
+			}
+			const accountId = isHex(address)
+				? '0x' + address.slice(4, -4)
+				: this.api.registry.createType('AccountId', address).toHex();
```

Comment:
1. Should the result of `isHex(address)` (which I assume has to loop over and check each character?) be cached at the top of the function to avoid calling it again here)?

2. Just trying to wrap my head around what's happening. So, if I provide arbitrary hex that happens to be one of the allowed lengths, `network` would still be `null`, `isValid` would be false, and the accountId would be some arbitrary bytes from my hex?

I guess since we return `isValid`, this is all good, but it makes me wonder whether the result type should really be either `{ isValid: true, ss58Prefix: 12, network: 'foo', accountId: '0x1234'}` or `{ isValid: false, ss58Prefix: null, network: null, accountId: null }`? Since if it's not valid, the other details make no sense?

Reply:
Thank you for the feedback @jsdw !
1. I am not sure if we need to cache the result of `isHex(address)` since from what I see [here](https://github.com/polkadot-js/common/blob/3597a4eb4e0cf9985104568058c5cd23701bbaed/packages/util/src/is/hex.ts#L25) it does not loop over every character. But if needed I will move it at the top.
2. You are right, it is more clear and correct to give all `null` values when `isValid` is `false`. I just changed it in my last commits. Very good point!

Reply:
1. Ah I see; I guess the current `isHex` would see something like `0xZZZZ` as valid hex, then! I'd probably cache the result to avoid the extra call to it but it's not a big deal given the current implementation!

Reply:
oh! Now I think I understand what you meant, to avoid calling twice the `isHex`. Yes, ofc, very good point. So I cache it in L40 and then I just check what boolean value the `addressIsHex` has.